packages/integrations/vercel/README.md

```diff
@@ -72,6 +72,43 @@ import vercel from '@astrojs/vercel/serverless';
 import vercel from '@astrojs/vercel/static';
 ```
 
+### Middleware
+
+You can use vercel middleware to get the request and redirect before the user loaded the page. Vercel middleware can run on both astro Edge, SSR or Static. You don't need to install `@vercel/edge` to access the middleware but you can install it to get feature such as geolocation. For more information about [middleware](https://vercel.com/docs/concepts/functions/edge-middleware)
+
+1. Configure your `astro.config.mjs`
+
+    ```js
+    // astro.config.mjs
+    import { defineConfig } from 'astro/config';
+    import vercel from '@astrojs/vercel/static';
+
+    export default defineConfig({
+      output: 'static', // 'server' or 'edge'
+      adapter: vercel(),
+    });
+    ```
```

Comment:
There’s nothing unique about this step right? It’s the same configuration you’d use when deploying without middleware? I’d suggest removing it in that case.

```suggestion
```